handler/src/test/java/io/netty/handler/logging/LoggingHandlerTest.java

```diff
@@ -270,18 +284,29 @@ public void shouldLogChannelReadComplete() throws Exception {
     private static final class RegexLogMatcher implements ArgumentMatcher<ILoggingEvent> {
 
         private final String expected;
+        private final boolean shouldContainNewline;
         private String actualMsg;
 
         RegexLogMatcher(String expected) {
+            this(expected, false);
+        }
+
+        RegexLogMatcher(String expected, boolean shouldContainNewline) {
             this.expected = expected;
+            this.shouldContainNewline = shouldContainNewline;
         }
 
         @Override
         @SuppressWarnings("DynamicRegexReplaceableByCompiledPattern")
         public boolean matches(ILoggingEvent actual) {
             // Match only the first line to skip the validation of hex-dump format.
             actualMsg = actual.getMessage().split("(?s)[\\r\\n]+")[0];
-            return actualMsg.matches(expected);
+            if (actualMsg.matches(expected)) {
+                // The presence of a newline implies a hex-dump was logged
+                return actual.getMessage().contains(NEWLINE) == shouldContainNewline; 
+            } else {
```

Comment:
nit: remove `else` as you return in the `if` block anyway.

Reply:
I like the alternate branches to have the same indentation, but will remove.